src/MarginTrading.Backend.Services/Services/CfdCalculatorService.cs

```diff
@@ -59,13 +61,64 @@ public decimal GetQuoteRateForQuoteAsset(string accountAssetId, string assetPair
             return 1.0M / quote.Ask;
         }
 
+        public decimal GetFplRate(string accountAssetId, string instrumentId, string legalEntity, bool fplSign)
+        {
+            var assetPair = _assetPairsCache.GetAssetPairById(instrumentId);
+            
+            if (accountAssetId == assetPair.QuoteAssetId)
+                return 1;
+
+            var assetPairQuoteAccount = _assetPairsCache.TryFindAssetPairStraight(assetPair.QuoteAssetId, accountAssetId, legalEntity);
```

Comment:
Complex and strange logic. Why just don't find pair and check if accountAssetId is quote or base?

Reply:
Fixed